downloads/swan-lake-release-notes/swan-lake-preview8/RELEASE_NOTE.md

```diff
@@ -0,0 +1,519 @@
+	int totalMarks = maths + physics + chemistry;
+	foreach [string, int?] [subject, marks] in grades.entries() {
+    	totalMarks = totalMarks + <int>marks;
+	}
+}
+```
+
+A named argument in a function call can correspond to an included record parameter in two different ways.
+
+1. Being relevant to a field of the included recorded parameter, which is not of the `never` type.
+
+    ```ballerina 
+    public function main() {
+        studentDetails(1001, "2020-Nov-20", firstName = "Peter", secondName = "So", age = 18);
+        studentDetails(1002, "2020-Nov-22", firstName = "Anne", secondName = "Doe", age = 20, address = {city: "Colombo", country: "Sri Lanka"});
+    }
+    ```
+
+2. Being relevant to a record rest descriptor of an included record parameter. Here, the field is described by a record rest descriptor either as explicitly part of an exclusive record type descriptor or as implicitly as part of an inclusive record type descriptor. The two conditions below should be satisfied for this.
+
+    - In the parameter list, there should be only one included record parameter with a record rest descriptor.
+    - The included record parameter with the record rest descriptor must include optional individual field descriptors of the type `never`, which correspond to each parameter name and the names of each individual field descriptor of the other included record parameters. In addition to these optional individual field descriptors, there should not be any other field descriptors with this included record parameter.
+
+    ```ballerina
+    public function main() {
+        studentTotalMarks(90, 85, 75);
+        studentTotalMarks(85, 85, 75, english = 75);
+        studentTotalMarks(75, 85, 90, english = 80, generalTest = 70);
+    }
+    ```
+
+##### Service Typing Changes
+
+Now, services are being based on objects. The service declaration syntax below is mere syntactic sugar for creating a new instance of a service class and then attaching it to a listener. With this change, the path that the service should serve on can be provided in the service-declaration syntax. This used to be provided using an annotation before.
+
+Therefore, the example below;
+
+```ballerina
+import ballerina/http;
+service hello on new http:Listener(9090) {
+
+    resource function sayHello() returns error? {
+
+    }
+}
+```
+
+will turn into the code below.
+
+```ballerina
+import ballerina/http;
+service [/optional_base_path] on new http:Listener(9090) {
+
+    resource function get sayHello() returns string {
+
+    }
+}
+```
+
+>**Note:** There is no longer a service variable name as `hello` in the first example. 
+Since services are objects, they can contain fields and regular methods. In addition to that, services can contain remote methods and resource methods.
+
+Resource functions are defined as follows:
+
+ `resource function` `resource-method-name` `resource-path``() { }`
```

Comment:
```suggestion
 `resource function` `resource-method-name` `resource-path() { }`
```

Reply:
I think I need to add a space between this two fragments, I want [resource-path] and [() { }] to appear in separate text spans.